include/kwk/detail/sequence/helpers.hpp

```diff
@@ -114,7 +114,7 @@ namespace kwk::__
   // Type short-cut for internal representation of types in axis
   //====================================================================================================================
   template<typename T>        struct stored_type : T {};
-  template<concepts::joker T> struct stored_type<T> { using type = std::int32_t; };
+  template<concepts::joker T> struct stored_type<T> { using type = std::uint16_t; };
```

Comment:
Not a fan to have that hardcoded. If you want it globally selectable instead of putting short everywhere in shape definition, I can live with a KWK_DEFAULT_DIMENSION_TYPE thing.

Reply:
right right, i have a lot of other changes incoming (still strugling with fully moving to new kiwaku, quite time consuming) so please wait a day or two until i push those - then will have to have another 'facetime' i'm pretty sure 🙈 😁

Reply:
actually this whole joker idea really 'broke my back', lost a lot of time on that - it made my code more complicated (as well as  the changes i required in kiwaku) - the previous approach where a dynamic/runtime-specified dimension/axis was specified simply with a zero/default-constructed value (using/counting on the fact/assumption that no one will ever need or use a fixed zero statically sized axis - what would be the point of that? i.e. such a shape will always have a zero count/volume)

Reply:
> actually this whole joker idea really 'broke my back', lost a lot of time on that - it made my code more complicated (as well as the changes i required in kiwaku) - the previous approach where a dynamic/runtime-specified dimension/axis was specified simply with a zero/default-constructed value (using/counting on the fact/assumption that no one will ever need or use a fixed zero statically sized axis - what would be the point of that? i.e. such a shape will always have a zero count/volume)

I needed something regular for the other algorithms and shape based operation to work with less complexity.

Reply:
> That's already available (see https://jfalcou.github.io/kiwaku/glossary.html#glossary-extent)
> 
> https://godbolt.org/z/ov5r59jh5

ok great in one sense but in the other it makes it even more complicated :/ for example now i have to detect the as<> thingy in addition to jokers as 'dynamic axes'...

i've already spent entire days on this but will/am now changeing joker back to your initial design (almost :D) and moving to as<> thingies...will see how it goes

Reply:
>I needed something regular for the other algorithms and shape based operation to work with less complexity.

it made my 'stuff' non-trivially more complex (still haven't cleaned up all the hacks around it) - i guess will have to do a live session to sort it out

Reply:
Yeah as I don't have a view on your use cases, I can't take them into account.
The last batch of changes was to accomodate the variadic nature of shape and stride and do the heterogeneous requirements. It has been done so that the thing was comprehensible and had a regular, predictible behavior. So, now, if your specs changed or isn't aligned with that, I need some more informations or we'll be chasing aroudn each other for years and we both have other stuff on our respective plates.